taichi/ir/type.cpp

```diff
@@ -166,12 +166,19 @@ bool QuantFloatType::get_is_signed() const {
   return digits_type_->as<QuantIntType>()->get_is_signed();
 }
 
-BitStructType::BitStructType(PrimitiveType *physical_type,
-                             std::vector<Type *> member_types,
-                             std::vector<int> member_bit_offsets)
+BitStructType::BitStructType(
+    PrimitiveType *physical_type,
+    const std::vector<Type *> &member_types,
+    const std::vector<int> &member_bit_offsets,
+    const std::vector<bool> &member_owns_shared_exponents,
+    const std::vector<int> &member_exponents,
+    const std::vector<std::vector<int>> &member_exponent_users)
     : physical_type_(physical_type),
       member_types_(member_types),
-      member_bit_offsets_(member_bit_offsets) {
+      member_bit_offsets_(member_bit_offsets),
+      member_owns_shared_exponents_(member_owns_shared_exponents),
+      member_exponents_(member_exponents),
+      member_exponent_users_(member_exponent_users) {
   TI_ASSERT(member_types_.size() == member_bit_offsets_.size());
```

Comment:
nit: do we need to make check more sizes given that we now have more vectors?

Reply:
done